src/domain/service/note.ts

```diff
@@ -63,7 +63,23 @@ export default class NoteService {
    * @param id - note internal id
    */
   public async deleteNoteById(id: NoteInternalId): Promise<boolean> {
-    return await this.noteRepository.deleteNoteById(id);
+    const hasRelation = await this.noteRelationsRepository.hasRelation(id);
+
+    if (hasRelation) {
+      const isNoteRelationsDeleted = await this.noteRelationsRepository.deleteNoteRelationsByNoteId(id);
+
+      if (isNoteRelationsDeleted === false) {
+        throw new DomainError(`Relation with noteId ${id} was not deleted`);
+      }
+    }
+
+    const isNoteDeleted = await this.noteRepository.deleteNoteById(id);
+
+    if (isNoteDeleted === false) {
+      throw new DomainError(`Note with id ${id} was not deleted`);
+    }
+
+    return isNoteDeleted;
```

Comment:
i tnink we should also delete team of currnet note, or in different PR?

Reply:
I think we should do this in a separate pull request so that we don't mix the logic of different features

Reply:
Create an issue please for deleting of Note Settings and Team